### PR TITLE
Deprecate `readandwrite` and add docs for `Pipe`

### PR DESCRIPTION
**Pull request by Keno**
Came up here: https://discourse.julialang.org/t/document-process-interaction-with-stdin-stout-stderr/7251/3.

Also document what `Pipe()` does, though note that this API may have to change due to #24717.

### CODE REVIEW

File: base/stream.jl [hunk 1]
```diff
@@ -552,6 +552,22 @@ mutable struct Pipe <: AbstractPipe
...
+Construct an uninitialized Pipe object.
+
+The appropriate end of the pipe will be automatically initialized if
+the object is used in process spawning. This can be useful to easily
+obtain references in process pipelines, e.g.:
+
+```
+julia> err = Pipe()
+
+# After this `p` will be initialized and you may read `foo`'s
```
Comment (mbaz) [inline base/stream.jl]: What is `p` here?

File: base/stream.jl [hunk 2]
```diff
@@ -552,6 +552,22 @@ mutable struct Pipe <: AbstractPipe
...
+# stderr from the `err` pipe.
+julia> spawn(pipeline(pipeline(`foo`, stderr=err), `cat`)
```
Comment (mbaz) [inline base/stream.jl]: The command is missing a closing )

Comment (mbaz) [reply base/stream.jl]: Personally I think we could also include an example with this syntax:

> spawn(`cat`, stdin, stdout, stderr)

Comment (mortenpi) [reply base/stream.jl]: The 4-argument `spawn()` appears to be [undocumented](https://docs.julialang.org/en/latest/stdlib/base/#Base.spawn) though.

**Comment by JeffBezanson** [comment]
Bump. Needs rebasing and doc string editing, and then should probably be merged for 1.0.

**Comment by stevengj** [comment]
@Keno, `p = open(cmd, "r+")` still doesn't seem to be mentioned in the documention; am I missing something?

Also, the apparent need to `close(p.in)` before reading (if the program is still waiting for input) is a bit fiddly (see also [this discourse thread](https://discourse.julialang.org/t/feeding-julia-strings-in-pipe-to-external-program/1333/7)).  Not sure if there is a better way.

In fact, the docstring for `open(cmd)` still uses the deprecated `open(command, stdio=devnull; write, read)` API.

**Pull request merged**